skyportal/tests/api/test_spectrum.py

```diff
@@ -742,3 +742,77 @@ def test_spectrum_external_reducer_and_observer(
     assert data['data']['observers'][0]['id'] == user.id
     assert data['data']['external_reducer'] == "Test external reducer"
     assert data['data']['external_observer'] == "Test external observer"
+
+
+def test_spectrum_plot_only_some_spectra(
+    upload_data_token, public_source, public_group, lris, sedm
+):
+    # upload spectra 1
+    status, data = api(
+        'POST',
+        'spectrum',
+        data={
+            'obj_id': str(public_source.id),
+            'observed_at': str(datetime.datetime.now()),
+            'instrument_id': lris.id,
+            'wavelengths': [664, 665, 666],
+            'fluxes': [234.2, 232.1, 235.3],
+            'group_ids': [public_group.id],
+        },
+        token=upload_data_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    # upload spectra 2
+    status, data = api(
+        'POST',
+        'spectrum',
+        data={
+            'obj_id': str(public_source.id),
+            'observed_at': str(datetime.datetime.now()),
+            'instrument_id': sedm.id,
+            'wavelengths': [664, 665, 666],
+            'fluxes': [134.2, 132.1, 135.3],
+            'group_ids': [public_group.id],
+        },
+        token=upload_data_token,
+    )
+
+    assert status == 200
+    assert data['status'] == 'success'
+    spec_id = data['data']['id']
+
+    status, data = api(
+        'GET', f'internal/plot/spectroscopy/{public_source.id}', token=upload_data_token
+    )
+
+    assert status == 200
+
+    # check how many different instruments are plotted
+    instruments = set()
+    refs = data['data']['bokehJSON']['doc']['roots']['references']
+    for r in refs:
+        if 'data' in r['attributes'] and 'instrument' in r['attributes']['data']:
+            instruments.add(r['attributes']['data']['instrument'][0])
+
+    assert len(instruments) > 1
+
+    # now ask for the same plot but with only one spectrum
+    status, data = api(
+        'GET',
+        f'internal/plot/spectroscopy/{public_source.id}',
+        params={'spectrumID': f'{spec_id}'},
```

Comment:
I'd change this call to have multiple IDs, so as to test the spliting functionality